fargate.tf

```
@@ -0,0 +1,9 @@
+module "fargate" {
+  source                            = "./modules/fargate"
+  cluster_name                      = aws_eks_cluster.this[0].name
+  profiles                          = var.create_eks ? var.eks_fargate_profiles : {}
```

```suggestion
  profiles   = var.eks_fargate_profiles
  create_eks = var.create_eks
```

For consistency I explained at https://github.com/terraform-aws-modules/terraform-aws-eks/pull/866#discussion_r427949268

---

I've removed the `create_eks_fargate` variable based on your other comment, and I think it made perfect sense to just check if any profiles are defined. The additional variable added no value and I don't think we should add it as `create_eks` again. The `create_eks` variable in the root module is still being honored by setting profiles = {} if it's false.

---

You're absolutely right, you can compute it with the `length` function. But, my point here is more about coding style which should match as much as possible with the rest of the module. So :

- it's more like a coding style consistency with the root and the first submodule
- it improve readability. No need for someone who's reading your code to try to compute it in his head. `create_eks` is `false` so it won't create any ressources, period.
- you don't need to compute it twice (even if it's a small thing here) when you're calling module and inside the module. You just pass the `create_eks` down to your resources

So I'll definitely prefer this writing in that context

```hcl
module "fargate" {
  source                            = "./modules/fargate"
  cluster_name                      = coalescelist(aws_eks_cluster.this[*].name, [""])[0]
  create_eks                        = var.create_eks
  fargate_profiles                  = var.fargate_profiles
  subnets                           = var.subnets
  tags                              = var.tags
  cluster_primary_security_group_id = element(concat(aws_eks_cluster.this[*].vpc_config[0].cluster_security_group_id, list("")), 0)
  worker_security_group_id          = local.worker_security_group_id
}
```